core/includes/backup/backup.mysql.inc

```diff
@@ -0,0 +1,583 @@
+    $out = '';
+    // Switch SQL mode to get rid of "CREATE ALGORITHM..." what requires more
+    // permissions + troubles with the DEFINER user.
+    $sql_mode = $this->query("SELECT @@SESSION.sql_mode")->fetchField();
+    $this->query("SET sql_mode = 'ANSI'");
+    $result = $this->query("SHOW CREATE VIEW `" . $view['name'] . "`", array(), array('fetch' => PDO::FETCH_ASSOC));
+    $this->query("SET SQL_mode = :mode", array(':mode' => $sql_mode));
+    foreach ($result as $create) {
+      $out .= "DROP VIEW IF EXISTS `" . $view['name'] . "`;\n";
+      $out .= "SET sql_mode = 'ANSI';\n";
+      $out .= strtr($create['Create View'], "\n", " ") . ";\n";
+      $out .= "SET sql_mode = '$sql_mode';\n";
+    }
+    return $out;
+  }
+
+  /**
+   * Get the SQL to insert the data for a given table.
+   */
+  protected function dumpTableDataSqlToFile(BackupFile $file, $table) {
+    $rows_per_query = $this->settings['data_rows_per_query'];
+    $rows_per_line = $this->settings['data_rows_per_line'];
+    $bytes_per_line = $this->settings['data_bytes_per_line'];
+
+    if ($this->settings['verbose']) {
+      $this->log('Table: %table', array('%table' => $table['name']), Backup::LOG_INFO);
+    }
+
+    // Escape backslashes, PHP code, special chars.
+    $search = array('\\', "'", "\x00", "\x0a", "\x0d", "\x1a");
+    $replace = array('\\\\', "''", '\0', '\n', '\r', '\Z');
+
+    $lines = 0;
+    $from = 0;
+    $args = array('fetch' => PDO::FETCH_ASSOC);
+    while ($data = $this->query("SELECT * FROM `" . $table['name'] . "`", array(), $args, $from, $rows_per_query)) {
+      if ($data->rowCount() == 0) {
+        break;
+      }
+
+      $rows = $bytes = 0;
+
+      foreach ($data as $row) {
+        $from++;
+
+        // DB Escape the values.
+        $items = array();
+        foreach ($row as $value) {
+          $items[] = is_null($value) ? "null" : "'" . str_replace($search, $replace, $value) . "'";
+        }
+
+        // If there is a row to be added.
+        if ($items) {
+          // Start a new line if we need to.
+          if ($rows == 0) {
+            $file->write("INSERT INTO `" . $table['name'] . "` VALUES ");
+            $bytes = $rows = 0;
+          }
+          // Otherwise add a comma to end the previous entry.
+          else {
+            $file->write(",");
+          }
+
+          // Write the data itself.
+          $sql = implode(',', $items);
+          $file->write('(' . $sql . ')');
+          $bytes += strlen($sql);
+          $rows++;
+
+          // Finish the last line if we've added enough items.
+          if ($rows >= $rows_per_line || $bytes >= $bytes_per_line) {
+            $file->write(";\n");
+            $lines++;
+            $bytes = $rows = 0;
+          }
+        }
+      }
+
+      // Finish any unfinished insert statements.
+      if ($rows > 0) {
+        $file->write(";\n");
+        $lines++;
+      }
+    }
+
+    if ($this->settings['verbose']) {
```

Comment:
Do we have a "verbose" setting, anyway? Do we need it? If so, we'd also need an equivalent for backup_migrate_get_peak_memory_usage(), as it's used below.

Reply:
All `backup_migrate_*`  usages are now eliminated.